src/Microsoft.IdentityModel.Protocols/Configuration/ConfigurationManager.cs

```diff
@@ -55,6 +65,16 @@ public class ConfigurationManager<T> : BaseConfigurationManager, IConfigurationM
         // refresh interval has passed.
         bool _refreshRequested;
 
+        private readonly AutoResetEvent _updateMetadata = new(false);
+        Task _updateMetadataTask;
```

Comment:
nit: this class/logic is more on a complex side, opting out here for more comments

```suggestion
         // Synchronization mechanism used to ensure that only one Task is updating the configuration.
        // By default, this AutoResetEvent is unsignaled.
        // The first and the following threads will wait until AutoResetEvent.Set is called first.       
        private readonly AutoResetEvent _updateMetadata = new(false);
        // Task that updates the configuration.
        Task _updateMetadataTask;
```